InMobiAdapter/src/main/java/com/chartboost/mediation/inmobiadapter/InMobiAdapter.kt

```diff
@@ -512,16 +553,37 @@ class InMobiAdapter : PartnerAdapter {
                 return Result.failure(ChartboostMediationAdException(ChartboostMediationError.CM_LOAD_FAILURE_INVALID_PARTNER_PLACEMENT))
             }
 
-            return suspendCoroutine { continuation ->
-                InMobiInterstitial(
-                    context,
-                    placement,
-                    buildFullScreenAdListener(
-                        request = request,
-                        partnerAdListener = partnerAdListener,
-                        continuation = continuation
+            return suspendCancellableCoroutine { continuation ->
+
+                inMobiInterstitialListeners[request.identifier] = buildFullScreenAdListener(
+                    request = request,
+                    partnerAdListener = partnerAdListener,
+                    continuation = continuation
+                )
+
+                // Get an already built interstitial ad listener and assign it to the hard reference.
+                inMobiInterstitialAdEventListener = inMobiInterstitialListeners.remove(request.identifier)
```

Comment:
imagine this scenario:
load placement A and it gets to line 562.
load placement B and then it gets to line 565.
what happens?

Reply:
placement B's overrides placement A's. Let me now push a change in which we just get the listener from its rightful key.